EmailToPrimaryContactBatch.cls

```diff
@@ -0,0 +1,161 @@
+/******************************************************************
+ * @Name         : EmailToPrimaryContactBatch
+ * @Description  : Batch Job of Email to Primary Contact
+ * @Created By   : Cagri Kilic
+ * @Created Date : Jan 12, 2023
+ * @Modification Log :
+ ******************************************************************
+ * Version        Developer        Date        Description
+ *------------------------------------------------------------
+ *
+ ******************************************************************/
+public without sharing class EmailToPrimaryContactBatch implements Database.Batchable<sObject>, Database.Stateful {
+    private Integer sentEmails = 0;
+    private Map<Id, Map<String, Decimal>> opportunityAmounts;
+    /******************************************************************
+  	* @Name         : start 
+  	* @Description  : Returns the result of the sent query
+  	* @Created By   : CagriKilic
+  	* @Created Date : Jan 12, 2023
+    * @Param bc     : Database.BatchableContext - Batchable Context
+    * @Return       : Database.QueryLocator - Query Locator
+
+  	******************************************************************/
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        try {
+            String query = 'SELECT StageName, Amount, AccountId FROM Opportunity';
+            return Database.getQueryLocator(query);
+        } catch (QueryException e) {
+            System.debug('An error occurred: ' + e.getMessage());
+            return null;
+        }
+    }
+    /******************************************************************
+  	* @Name         : execute 
+  	* @Description  : Processes data from query
+  	* @Created By   : CagriKilic
+  	* @Created Date : Jan 12, 2023
+    * @Param bc     : Database.BatchableContext - Batchable Context
+    * @Param con    : List<Opportunity> - List of Opportunities 
+    
+  	******************************************************************/
+    public void execute(Database.BatchableContext bc, List<Opportunity> opps) {
+        try {
+            opportunityAmounts = new Map<Id, Map<String, Decimal>>();
+            String stageName;
+            for (Opportunity o : opps) {
+                if (o.StageName != 'Closed Lost' && o.StageName != 'Closed Won') {
```

Comment:
please create a constan for closed lost/won (maybe a set and then check if set contains Stagename - it is closed, if no - open?)